03_letters/undrewb/main_test.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"bytes"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	want := strconv.Quote("a:2\nb:1\n")
+	got := strconv.Quote(buf.String())
+
+	if got != want {
+		t.Errorf("actual: %s does not match expected: %s", got, want)
+	}
+}
+
+var sortLettersTestData = []struct {
+	name  string
+	input string
+	want  string
+}{
+	{name: "lab example", input: "aba", want: "a:2,b:1"},
+	{name: "trailing newline", input: "aba\n", want: "\n:1,a:2,b:1"},
+	{name: "duplicate entries", input: "\t\t\\aba\n", want: "\t:2,\n:1,\\:1,a:2,b:1"},
```

Comment:
I understand that you are trying to achieve good test coverage, but I think this might be a little bit of overkill. It's important to consider what each test is trying to achieve when building out test cases. I feel that a lot of these cases are kind of testing the same thing (`one repeated letter`, `random example`, `whitespace`, `duplicate entries`, `lab example`) and many of these can be removed. 

If you think about structuring your tests and having each test try to achieve / verify something in particular, you should be able to hit most of these cases with a couple of tests.

 * Empty input -> Corner case
 * Multiple letters (different languages if you want) -> Positive Case
 * Emoji input -> Verify data structure is using rune